[CCP] Implemented compare treatment feature for outcomes table and icons

noranda opened the pull request:
This PR satisfies JIRA task [2105](https://jira.mitre.org/browse/ASCODCP-2105) and implements the compare treatment feature for both the outcomes table and the icons views as [designed](https://www.figma.com/proto/7y2xB3nMuEuWIwWlgjejZi/CCP-cancerlinq-limited?node-id=555%3A7183&viewport=1190%2C741%2C0.5&scaling=min-zoom).

- Added redux action and reducer for setting selected treatment used in both table and icons views
- Added treatment selection for table view
- Added treatment selection and treatment display for icons view
- Changed `.js` components to `.jsx` for consistency
- Styled new features and fixed responsive styles for both the table and icons view
- Fixed and added tests

_Pull requests into Flux require the following. Submitter should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:. Reviewers should complete every item on the bulleted list when reviewing._


## Submitter:

**Running the Application**

- [x] Manually tested in Chrome (primary browser for testing)
- [x] Manually tested in IE/Safari (verify app loads and basic feature testing)
- [x] Manually tested in Chrome in iPad mode (at minimum, viewing patient record)

**Task**

- [x] Branch implements task as expected
- [ ] Key application features are updated as needed for task and tested (see: [Key Application Features](https://github.com/FluxNotes/flux/wiki/Key-Application-Features))
- [ ] Task specific definition of done (as described in the Jira task) is met, if applicable.

**Documentation**

- [x] This pull request describes why these changes were made
- [x] Recognizes any potential shortcomings/bugs in the description 
- [x] Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
- [ ] Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
- [ ] Document any new APIs/extension points
- [ ] Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**

- [x] Code style guide is followed (see: [Code Style Guide](https://github.com/FluxNotes/flux/wiki/JavaScript-and-HTML-Code-Style-Guide))
- [x] Code is commented

**Tests**

- [x] Existing tests passed
- [x] Added backend tests for new functionality added


## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code

dreeves4321 commented:
Looking good!
Several thoughts:
[DR1] The titles in the key don't really work in the current context. Maybe "decrease/increase in survival rate due to change in treatment." Or simply "increase in survival rate"
[DR2] The backing for the red/gray stripes should be white, so that the survival (gray) bar stops at the correct location.
[DR3] The columns in the title row aren't aligned with the body.
[DR4] In icons, for treatments with FEWER survive, there should be red borders on the gray circles associated with that loss.
[DR5] "3 less survive..." should be "3 fewer survive..."

dreeves4321 commented:
My last thought is I'm not very happy with the selection interaction. I feel we need some hover interactions to help clarify a bit. I'll work out something and get back to you Noranda.

dreeves4321 commented:
[DR6] you can deselect a row in the icon view. I don't think that's necessary, get rid of that (but maybe I'm missing something?)

dreeves4321 commented:
I do like the short selection animation when you select a row. Is there something similar we can apply when selecting a comparison? I bet sliding rows is hard, but maybe there's a pulse we can apply to the comparison icon that, if timed right, would help tie the selection action to the change in the top row... any thoughts?

noranda commented on src/mcode-pilot/components/TreatmentOptionsOutcomesIcons/TreatmentOptionsOutcomesIcons.jsx:
```diff
@@ -0,0 +1,195 @@
+                    <div className="display-name">{treatment.displayName}</div>
+                </div>
+
+                <div className="flex-1 flex-padding total-patients">({normalizedTotalPatients})</div>
```
I think the point of the icons view is to represent the data out of 100 patients, which carries over to the data, so stating these are percentages is redundant. This could probably be made a little more clear, though. @dreeves4321 Thoughts?

dreeves4321 replied:
if this is the column with total number of patients, I agree, it should be clearer as # out of 100 patients. Maybe format the same way as the survival column (33/100)?

noranda replied:
Done!

noranda commented on src/mcode-pilot/components/TreatmentOptionsOutcomesIcons/TreatmentOptionsOutcomesIcons.jsx:
```diff
@@ -0,0 +1,195 @@
+                    {this.renderInteractionHeader()}
+                    {selectedTreatmentInData && this.renderInteractionRow(selectedTreatment, true)}
+                    {similarPatientTreatmentsData.map(treatmentData => {
+                        if (!selectedTreatment || treatmentData.displayName !== selectedTreatment.displayName) {
```
Fixed! I was originally checking id's until I realized the id's are not consistent, so switched to display names, but missed this one. Thanks!

noranda commented:
@KeeyanGhoreshi:
[KG1]: fixed

@dreeves4321:
[DR1]: fixed
[DR2]: removed
[DR3]: fixed
[DR4]: fixed
[DR5]: fixed
[DR6]: fixed -- I was on the fence about this one anyways :)
Added the hover changes we discussed.
Regarding animations, I actually didn't put any in these components. If you think we should add them, I'd suggest making a new task for that.

Thanks!

dreeves4321 commented:
Looks good!
